js/server.js

```diff
@@ -19,117 +18,119 @@ const Utils = require("./utils.js");
  * Server
  *
  * @param {object} config The MM config
- * @param {Function} callback Function called when done.
  * @class
  */
-function Server(config, callback) {
+function Server(config) {
+	const app = express();
 	const port = process.env.MM_PORT || config.port;
 	const serverSockets = new Set();
-
 	let server = null;
-	if (config.useHttps) {
-		const options = {
-			key: fs.readFileSync(config.httpsPrivateKey),
-			cert: fs.readFileSync(config.httpsCertificate)
-		};
-		server = require("https").Server(options, app);
-	} else {
-		server = require("http").Server(app);
-	}
-	const io = require("socket.io")(server, {
-		cors: {
-			origin: /.*$/,
-			credentials: true
-		},
-		allowEIO3: true
-	});
-
-	server.on("connection", (socket) => {
-		serverSockets.add(socket);
-		socket.on("close", () => {
-			serverSockets.delete(socket);
+
+	this.open = function () {
```

Comment:
I think this should have the callback. I'll need it in https://github.com/MichMich/MagicMirror/pull/2952 and should probably wait for `on("listening")` before calling back (I'll fix that in that PR if I can get the tests to work).

Reply:
so, you approve of this PR anyway?

Reply:
yeah :)

I'll add it back if I need it.